experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricReader.ts

```diff
@@ -14,52 +14,140 @@
  * limitations under the License.
  */
 
+import * as api from '@opentelemetry/api';
 import { AggregationTemporality } from './AggregationTemporality';
-import { MetricExporter } from './MetricExporter';
 import { MetricProducer } from './MetricProducer';
+import { MetricData } from './MetricData';
+import { callWithTimeout } from '../utils';
+
+export type ReaderOptions = {
+  timeoutMillis?: number
+}
+
+export type ReaderCollectionOptions = ReaderOptions;
+
+export type ReaderShutdownOptions = ReaderOptions;
+
+export type ReaderForceFlushOptions = ReaderOptions;
 
 // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricreader
 
+/**
+ * A registered reader of metrics that, when linked to a {@link MetricProducer}, offers global
+ * control over metrics.
+ */
 export abstract class MetricReader {
+  // Tracks the shutdown state.
   private _shutdown = false;
+  // MetricProducer used by this instance.
   private _metricProducer?: MetricProducer;
 
-  constructor(private _exporter: MetricExporter) {}
+  constructor(private readonly _preferredAggregationTemporality = AggregationTemporality.CUMULATIVE) {
+  }
 
+  /**
+   * Set the {@link MetricProducer} used by this instance.
+   *
+   * @param metricProducer
+   */
   setMetricProducer(metricProducer: MetricProducer) {
```

Comment:
Ah, is this taking the Java route of passing a metric-producer rather than hard-coding `collect` to talk to static registry of metrics.

I'm wondering if we should update the specification so that this is the "default" specified mechanism and the .NET hook is an alternative option to make it more clear to implementers that wish to a hard-link between metric-reader implementation + the sdk.   Did you find this confusing at all?

Reply:
It is confusing on the spec side. I've previously submitted issues like https://github.com/open-telemetry/opentelemetry-specification/issues/2072 for it.

Reply:
I agree that it is somewhat confusing on the spec side, when I was reading it the same question described in the issue above popped into my mind. I think updating the spec would help in that regard.